docs/design/proposals/codegen-refactor/model_cmd_2.md

```diff
@@ -0,0 +1,143 @@
+* Long dev loop; after making changes to either *generator.yaml* or logic in *code-generator* users need to either manually verify files or run e2e tests-- both of which are time consuming today
+* How *code-generator* reconciles `ackgenconfig` with `aws-sdk` to create `ackmodel` is inconsistent (and some duplicated work). For example, overrides such as `custom_shapes` [edit aws-sdk directly](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/sdk/custom_shapes.go#L62) while others such as `ignored_operations` [set operation values to nil](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/pkg/model/model.go#L295) in memory leaving the actual operations in `aws-sdk` untouched.
+
+Current Pipeline
+---
+
+![current-pipeline](./images/current_pipeline.png)
+* Displayed are the important and mostly duplicated calls between the `apis` and `controller` commands
+* The cache is only used to store the `aws-sdk` repo
+* The red line denotes a dependency on $SERVICE-controller (output directory) repo that should be removed
+* Only `GetCRDs` is duplicated between the commands when creating the model
+
+
+Current Dev Loop
+---
+
+![current-dev-loop](./images/current_dev_loop.png)
+* Validating the result of a change in *code-generator* and/or *generator.yaml* can be a lengthy process
+* In the current feedback loop, users and contributors need to manually verify files or run lengthy e2e tests even for minor changes to `ackgenconfig`
+
+
+## Solution
+Add a new command, `model`, which takes in `aws-sdk` and *generator.yaml* as inputs to create and cache `ackmodel`. Existing commands, `api` and `controller`, will remove parameters for `aws-sdk` and *generator.yaml* and replace them with `ackmodel` input and/or let it default to `--cache-dir`.
+
+Updated Pipeline
+---
+![updated-pipeline](./images/proposed_pipeline.png)
+* This is not an exhaustive diagram of the calls, but shows the clear responsibility of `generateModel` and how downstream commands like `api` and `controller` become significantly lighter and easier to follow.
+* `ackmodel` will be cached in the same folder as `aws-sdk` and downstream commands will check for `ackmodel` instead of `aws-sdk`
+* users can override with a path to their own `ackmodel` (for even quicker testing!), but by default will check `--cach-dir`
+* No more dependency on $SERVICE-controller repo. The `ackmodel` will be generated with a specific version which can be extracted after commands `GetACKModel()`
+
+Updated Dev Loop:
+---
+![updated-dev-loop](./images/proposed_dev_loop.png)
+* Diffing the `ackmodel` between *code-generator* runs will be quicker than manually checking files or running e2e tests
```

Comment:
Good point. I think it depends on what changes/configs are being edited, but for the most part I would agree that comparing `ackmodel`s won't necessarily be easier (at least today). One example where I do think it's easier would be `output_wrapper_field_path` addition/removal. Given we edit `aws-sdk` directly in `ackmodel` (per 2nd proposal), it would be a lot easier to do a diff between `ackmodel.sdkapi` before/after `./ack-generate model` runs to see whether or not a shape is being unwrapped properly vs. today's approach where we not only diff files in `$SERVICE-controller` but also manually checking for syntax errors/incorrect types being used.

Overall, comparing `ackmodel` will probably start having more benefits in the future especially as we simplify the interface in `generator.yaml` and push down complexity. Adding a dev tool to help compare `ackmodel` between runs could also be a possibility and will result in easier debugging and quicker feedback loops for devs.